Intersect.Client/Maps/MapInstance.cs

```diff
@@ -1358,6 +1355,17 @@ public void Dispose(bool prep = true, bool killentities = true)
             Delete();
         }
 
+        public static bool MapNotRequested(Guid mapId) => !MapRequests.ContainsKey(mapId) || MapRequests[mapId] < Timing.Global.Milliseconds;
+
+        public static void UpdateMapRequestTime(Guid mapId)
+        {
+            if (MapRequests.ContainsKey(mapId))
+            {
+                MapRequests[mapId] = Timing.Global.Milliseconds + 2000;
```

Comment:
you actually only need this as the only line in this method -- indexers will add the key if it does not exist, ContainsKey not needed